visualization/app/codeCharta/codeCharta.component.ts

```diff
@@ -75,10 +76,10 @@ export class CodeChartaController {
 			.then(() => {
 				this.storeService.dispatch(setState(ScenarioHelper.getDefaultScenarioSetting()))
 			})
-			.catch(e => {
+			.catch((errors: ErrorObject[]) => {
```

Comment:
It is not a good idea to reject arrays. Please always only reject / throw actual error.

Note: while working with promises it is always recommended to use async functions. Those do not only improve readability but also provide good stack traces.